src/utils/appEngagementMonitor/appEngagementMonitor.js

```diff
@@ -0,0 +1,30 @@
+const { query: Hasura } = require('../hasura');
+
+const userInactivityTracker = async () => {
+  try {
+    // TODO: Fetch user_actions data from the DB
+
+    for (const user of users.rows) {
+      const { user_id, last_app_opened_timestamp } = user;
+
+      const today = new Date();
+      const appOpenedToday = last_app_opened_timestamp >= today.setHours(0, 0, 0, 0) && last_app_opened_timestamp <= today.setHours(23, 59, 59, 999);
+
+      if (appOpenedToday) {
+        // TODO: insert the activity award points to the user
+      } else {
+        const lastOpenedDate = new Date(last_app_opened_timestamp);
+        const timeDiff = today - lastOpenedDate;
+        const consecutiveInactiveDays = Math.floor(timeDiff / (1000 * 60 * 60 * 24));
+
+        if (consecutiveInactiveDays >= 5) {
```

Comment:
This will deduct points everyday after the 5th day, we want to deduct points only at every 5th day. I.e., we should deduct the points only on the 5th, 10th, 15th... days